seatunnel-e2e/seatunnel-connector-v2-e2e/connector-kafka-e2e/src/test/resources/avro/kafka_avro_to_assert.conf

```diff
@@ -82,7 +82,21 @@ sink {
                 rule_value = 99
               }
             ]
-          }
+          },
+          {
+           field_name = c_string
+           field_type = string
+           field_value = [
+            {
+              rule_type = MIN_LENGTH
+              rule_value = 6
+            },
+            {
+              rule_type = MAX_LENGTH
+              rule_value = 6
+            }
+          ]
+         }
```

Comment:
I still hope to do an end-to-end test on all fields to ensure that the data is not only consistent in the amount of data, but also to ensure that each data and field are consistent, and it is best to write the data to the database for a Check
 cc @hailin0 @Hisoka-X

Reply:
assert sink can't do some equal check, so i add the data check [in code](https://github.com/apache/seatunnel/pull/6082/files#diff-5066134a42342f0962a0d21bce0ec987a077faf189d92cf0854f516f90a8ad21R418)

Reply:
> assert sink can't do some equal check, so i add the data check [in code](https://github.com/apache/seatunnel/pull/6082/files#diff-5066134a42342f0962a0d21bce0ec987a077faf189d92cf0854f516f90a8ad21R418)

But you can write it to a database such as pg and then read it to make sure the input and output are consistent,The type is only one aspect of the test and we must make sure that the data values are exactly the same

Reply:
yeah, so i add the data check after job finish, read data from kafka topic and verify the data consistent

Reply:
> yeah, so i add the data check after job finish, read data from kafka topic and verify the data consistent

ok for me.

Reply:
Hi @zhilinli123  I update the code. 
In this update, it can verify all field value and type. also can test the serialize (ST row -> Avro Generic Record -> bytes ) and deserialize (bytes -> Avro Generic Record -> ST Row ). I think this test can cover this feature.